modules/porous_flow/src/materials/PorousFlowPermeabilityTensorFromVar.C

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowPermeabilityTensorFromVar.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowPermeabilityTensorFromVar);
+
+template <>
+InputParameters
+validParams<PorousFlowPermeabilityTensorFromVar>()
+{
+  InputParameters params = validParams<PorousFlowPermeabilityBase>();
+  params.addRequiredCoupledVar("perm", "The scalar permeability");
+  params.addParam<RealTensorValue>("k_anisotropy",
+                                   "A tensor to multiply the scalar "
+                                   "permeability, in order to obtain anisotropy if "
+                                   "required. Defaults to isotropic permeability "
+                                   "if not specified.");
+  params.addClassDescription(
+      "This Material calculates the permeability tensor from a coupled variable "
+      "multiplied by a tensor");
+  return params;
+}
+
+PorousFlowPermeabilityTensorFromVar::PorousFlowPermeabilityTensorFromVar(
+    const InputParameters & parameters)
+  : PorousFlowPermeabilityBase(parameters),
+    _perm(coupledValue("perm")),
+    _k_anisotropy(parameters.isParamValid("k_anisotropy")
+                      ? getParam<RealTensorValue>("k_anisotropy")
+                      : RealTensorValue(1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0))
```

Comment:
Suggestion: This is fine, but you can make this a little more compact by supply the default right in the declaration of the parameter. Then you don't have to have a conditional here where you "get" the value.